src/NLog.Targets.Syslog/MessageCreation/Rfc3164.cs

```diff
@@ -22,13 +22,15 @@ internal class Rfc3164 : MessageBuilder
         private readonly TagPolicySet tagPolicySet;
         private readonly PlainContentPolicySet plainContentPolicySet;
         private readonly AsciiMessagePolicy asciiMessagePolicy;
+        private readonly ASCIIEncoding asciiEncoding;
 
         public Rfc3164(Facility facility, LogLevelSeverityConfig logLevelSeverityConfig, Rfc3164Config rfc3164Config, EnforcementConfig enforcementConfig) : base(facility, logLevelSeverityConfig, enforcementConfig)
         {
             hostnamePolicySet = new PlainHostnamePolicySet(enforcementConfig);
             tagPolicySet = new TagPolicySet(enforcementConfig);
             plainContentPolicySet = new PlainContentPolicySet(enforcementConfig);
             asciiMessagePolicy = new AsciiMessagePolicy(enforcementConfig);
+            asciiEncoding = new ASCIIEncoding();
```

Comment:
Can you explain more about this thread safety issue? As we should call Microsoft and tell about this. Since it is very common to just reference [System.Encoding.UTF8](https://docs.microsoft.com/en-us/dotnet/api/system.text.encoding.utf8) instead of allocating own Encoding-instances.

Reply:
`Encoding`, `AsciiEncoding` and `UTF8Encoding` docs used to say the any instance members are not guaranteed to be thread safe
Documentation does not contain that statement anymore and [it seems](https://stackoverflow.com/questions/3024384/thread-safety-and-system-text-encoding-in-c-sharp) that, regardless of documentation, they are implemented as thread-safe.

It would be better to open an issue on this since I need to review the encoding usage across the solution

Reply:
Removed optimization.